newIDE/app/src/SceneEditor/index.js

```diff
@@ -626,18 +676,32 @@ export default class SceneEditor extends Component {
           }}
         />
         <ObjectEditorDialog
-          open={!!this.state.editedObject}
-          object={this.state.editedObject}
+          open={!!this.state.editedObjectWithContext.object}
+          object={this.state.editedObjectWithContext.object}
           project={project}
           resourceSources={resourceSources}
           resourceExternalEditors={resourceExternalEditors}
           onChooseResource={onChooseResource}
           onCancel={() => {
-            this.reloadResourcesFor(this.state.editedObject);
+            this.reloadResourcesFor(this.state.editedObjectWithContext.object);
             this.editObject(null);
           }}
+          canRenameObject={newName => {
+            if (
+              this._canObjectUseNewName(
+                this.state.editedObjectWithContext,
+                newName
+              )
+            ) {
+              this.setState({ editedObjectNewName: newName });
```

Comment:
Having a `this.setState` in a function that is named `canBlabla` is a good way to see that's something can be going wrong here (as we expect a function called `canRenameObject` to only do a check, not to change a state somewhere) :) 
1) Either the function is in charge of updating the state, meaning that SceneEditor owns the name being edited. In which case you rename this function `onChangeObjectName`.
OR 2) you remove the this.setState (and the name make sense now!) and use `onApply` to have the dialog pass to the parent the final name that was chosen.

I would go for 2 if possible :) (as you already made a state in ObjectsEditorDialog, and make more sense to me to give the responsibility of holding the new name of the object in the dialog, until we click on Apply as which moment the parent takes care of doing the renaming).

Reply:
Done, comit coming later today